### DISCUSSION TITLE
How to wait for a request by URL in Rod

### DISCUSSION THREAD

**Discussion by davidberetta** [go-rod/rod#1071] [Q&A] [answered]
Hi, I am new to golang and rod. I am coming from a nodejs background where I have used puppeteer extensively. I would like to know how to wait for a specific response URL to load after navigation and capture the response body and headers. In puppeteer I would do something like this:

`const tokenResponse = await this.page.waitForResponse((res: HTTPResponse) => res.url().includes('token'), { timeout: 5000 });`

**Comment by ysmood** [comment]
You can use `page.WaitEvent` to do similar job: https://go-rod.github.io/#/events/README

**Reply by davidberetta** [reply]
Hi @ysmood, thank you for your reply. I tried this approach but it does not give me what I want. It does not seem to capture the event that I am needing. When the page navigates it makes several XHR requests and I am trying to capture one of those requests and get the response headers and body. Is there a way to do this with page.WaitEvent?

I am not sure if this is the correct approach, but I was able to get the response using browser.HijackRequests()

```
	router := browser.HijackRequests()
	defer router.MustStop()

	router.MustAdd("*/token", func(ctx *rod.Hijack) {

		ctx.MustLoadResponse()
		body := ctx.Response.Body()

		j := TokenResp{}
		json.Unmarshal([]byte(body), &j)

		log.Println("Access Token: ", j.AccessToken)
	})

	go router.Run()
```

**Reply by ysmood** [reply, marked as answer]
Then you need hijack, since you also want the body.
